loadtest/ff_gen.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+const (
+	BOOL   = "boolean"
+	STRING = "string"
+)
+
+/*
+A simple random feature flag generator for testing purposes. Output is saved to "random.json".
+
+Configurable options:
+
+	-c : feature flag count (ex:go run ff_gen.go -c 500)
+	-t : type of feature flag (ex:go run ff_gen.go -t string). Support "boolean" and "string"
+*/
+func main() {
+	// Get flag count
+	var flagCount int
+	flag.IntVar(&flagCount, "c", 100, "Number of flags to generate")
+
+	// Get flag type : Boolean, String
+	var flagType string
+	flag.StringVar(&flagType, "t", BOOL, "Type of flags to generate")
+
+	flag.Parse()
+
+	if flagType != STRING && flagType != BOOL {
+		fmt.Printf("Invalid type %s. Falling back to default %s", flagType, BOOL)
+		flagType = BOOL
+	}
+
+	root := Flags{}
+	root.Flags = make(map[string]Flag)
+
+	switch flagType {
+	case BOOL:
+		root.setStringFlags(flagCount)
```

Comment:
Without diving too far into this myself, these two cases seem reversed. Bool sets string, strings sets bool. Is that correct?

Reply:
@tcarrio  thanks. yes this is a mix in the final refactoring (extracting constants to consts). Fixed with the latest commit.

Reply:
Further, this does not have any impact on the results as results were obtained from the script version preset at 07f833b56eacd0870099e53b5f07b201478d8625